cli-cmd/account: add new files and change some related packages.

Doresimon opened the pull request:
# modify some functions to unique package.

1. func: password enter
2. func: add account
3. func: set account
4. func: delete account
5. func: encrypt account file

# refactor types from account.go to account_cmd.go

# add -label flag for <account add> <account set>
* <account list> will show label info at the same time
* default label for an account is ""(blank string)

# add <account import> sub command
* account import -source xxx: to import accounts in xxx to current wallet file.
* use -file <filename> to specify current wallet file. default name is "wallet.dat". 

# change some print info.

Signed-off-by: Halo Guan <guanjixing@onchain.com>

JasonZhouPW committed:
add 'succeed' field in result json

JasonZhouPW committed:
modify checkWitness function

JasonZhouPW committed:
add vmcode to wasm vm

JasonZhouPW committed:
tmp commit for add smartcontract apis

JasonZhouPW committed:
add neovm apis to wasmvm

JasonZhouPW committed:
replace errors with ontology errors

JasonZhouPW committed:
fix json unmarshal bug and refactor code

JasonZhouPW committed:
add call neovm functions

Doresimon committed:
cmd/account: add new files and change some related packages.

# modify some functions to unique package.

1. func: password enter
2. func: add account
3. func: set account
4. func: delete account
5. func: encrypt account file

# change some print info.

Signed-off-by: Halo Guan <guanjixing@onchain.com>

Doresimon committed:
use common/password for account_cmd, remove old password enter function.

Doresimon committed:
change some functions' logic

JasonZhouPW committed:
fix callcontract call wasm vm result is nil bug

laizy commented on cmd/account_cmd.go:
```diff
@@ -321,12 +314,12 @@ func encrypt(ctx *cli.Context) error {
 
 	//let user enter password with double check
-	pass := password.DoubleEnterPassword()
+	pass, _ := password.GetConfirmedPassword()
```
handle all err in this file.

laizy commented on account/account.go:
```diff
@@ -106,3 +109,132 @@ func (ac *Account) PubKey() keypair.PublicKey {
+
+	"SM2P256V1": {"SM2P256V1", keypair.SM2P256V1},
+	"ED25519":   {"ED25519", keypair.ED25519},
+}
```
so why move all those maps here? account module does not care any of those stuff. they are just used to map user's  command line input.  so keep it untouched should be a better choice.

lightshine001 committed:
Cleanup p2p code (#96)

Signed-off-by: liupan@onchain.com <liupan@onchain.com>

Commit:
fixed invalid back slash use in docs.

Signed-off-by: junjie <wengjunjie@onchain.com>

Commit:
replace RLock with Lock  (#100)

signed-off-by:wangbing@onchain.com

Arbio5zt committed:
update get public key cmd with new cli. (#102)

Signed-off-by: luodanwg <luodan.wg@gmail.com>

Arbio5zt committed:
fixed misspell (#103)


Signed-off-by: junjie <wengjunjie@onchain.com>

laizy committed:
fix typos: statefull -> stateful (#101)

Signed-off-by: laizy <laizhichao@onchain.com>

laizy committed:
remove redundant NodeCnt field in p2p Addr message (#95)

Signed-off-by: laizy <laizhichao@onchain.com>

Commit:
fix min failed node error bug (#104)

signed-off-by:wangbing@onchain.com

Doresimon committed:
add -label flag to <account add> <account set>
> add ... -label XXX: to add an account labeled XXX
> set ... -label XXX: to set an account's label to XXX
> list ...: will show account's label too

Signed-off-by: Halo Guan guanjixing@onchain.com

Doresimon committed:
add <account import> command
> import -file,-f <filename>:  specify the current wallet file, default is wallet.dat
> import -source,-s <filename>: specify the source file to import.

Signed-off-by: Halo Guan guanjixing@onchain.com

Doresimon committed:
remove remarks.

Signed-off-by: Halo Guan <guanjixing@onchain.com>

Doresimon committed:
Merge remote-tracking branch 'origin/master'
